manticore/ethereum.py

```diff
@@ -1078,6 +1123,38 @@ def global_findings(self):
     def workspace(self):
         return self._executor._workspace._store.uri
 
+    def _concretize_offsets_and_sizes(self, state, signature, data):
+        ''' Using signature spec this function browse the data and concretize 
+            all the offsets to dynamic arguments and all the size of the dynamic
+            arguments so it all fits into data.
+            The available space is fairly divided among all the dynamic arguments.
+        '''
+        is_multiple, func_name, types = ABI.parse_signature_spec(signature)
+        dyn_arguments = []
+        for pos, t in enumerate(types):
+            dynamic = t in (u'bytes', u'string') or t.endswith(']')
+            if dynamic:
+                dyn_arguments.append(pos)
+
+        used = 4 + len(types)*32 
+        available = len(data) -used -len(dyn_arguments)*32
+
+        for i in range(0, len(dyn_arguments)):
+            #get, constraint and concretize dyn_offset to some reasonable value
+            dyn_offset = ABI.get_uint(data, 4 + dyn_arguments[i]*32, 256)
+            assert solver.can_be_true(state.constraints, dyn_offset == used)
+            state.constrain(dyn_offset == used)
```

Comment:
we concretize dyn_offset by adding a constraint to a constant value, and we concretize `data[used:used+32]` (the size of the dynamic element) by directly assigning it.

shouldn't we prefer the former method? when we do the latter, we are erasing other data. that other data potentially could have been symbolic data with constraints on it?